subprojects/build-cache-http/src/main/java/org/gradle/caching/http/internal/DefaultHttpBuildCacheServiceFactory.java

```diff
@@ -44,25 +45,36 @@ public DefaultHttpBuildCacheServiceFactory(SslContextFactory sslContextFactory)
     }
 
     @Override
-    public BuildCacheService createBuildCacheService(HttpBuildCache configuration) {
+    public BuildCacheService createBuildCacheService(HttpBuildCache configuration, Describer describer) {
         URI url = configuration.getUrl();
         if (url == null) {
             throw new IllegalStateException("HTTP build cache has no URL configured");
         }
+        URI noUserInfoUrl = stripUserInfo(url);
 
         Collection<Authentication> authentications = Collections.emptyList();
         if (configuration.getCredentials().getUsername() != null && configuration.getCredentials().getPassword() != null) {
-            try {
-                url = new URI(url.getScheme(), null, url.getHost(), url.getPort(), url.getPath(), url.getQuery(), url.getFragment());
-            } catch (URISyntaxException e) {
-                throw new GradleException("Error constructing URL for http build cache", e);
-            }
+            url = noUserInfoUrl;
             DefaultBasicAuthentication basicAuthentication = new DefaultBasicAuthentication("basic");
             basicAuthentication.setCredentials(configuration.getCredentials());
             authentications = Collections.<Authentication>singleton(basicAuthentication);
         }
 
+        describer.type("HTTP").config("url", noUserInfoUrl.toASCIIString());
+        if (!authentications.isEmpty() || url.getUserInfo() != null) {
+            describer.config("authenticated", null);
+        }
```

Comment:
Would it make sense for this to be symmetrical with/without authentication?
```
describer.config("authenticated", !authentications.isEmpty() || url.getUserInfo() != null);
```

So when there is no authentication, we have `authenticated=false` and when there is, we have `authenticated=true`.